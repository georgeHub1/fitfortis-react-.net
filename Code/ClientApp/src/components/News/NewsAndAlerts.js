import React, { Component } from 'react';
import { Tabs } from 'antd';
import { FormattedMessage, injectIntl } from 'react-intl';
import NewsAndAlertsNews from './NewsAndAlertsNews';
import NewsAndAlertsAlerts from './NewsAndAlertsAlerts';
import styles from './styles.module.less';
import { localeSet } from '../../redux/locale.action';
import { connect } from 'react-redux';
import analyticId from '../../utils/AnalyticsAndAbTests.tsx';
const { TabPane } = Tabs;

class NewsAndAlerts extends Component {
  displayName = NewsAndAlerts.name

  componentDidMount () {
    if (window.location.search.includes('?lang=')) {
      const lan = window.location.search.includes('?lang=') ? window.location.search.substring(6, 11) : (localStorage.getItem('language') || 'en');

      this.props.dispatch(localeSet(lan));
      localStorage.setItem('language', lan);
      analyticId.firebaseAnalyticsLog('Alerts');
    }
  }
  render () {
    return (
      <div className="newsAndAlertsPage">
        <h1 className={styles.header}>
          <FormattedMessage id="NewsAndAlerts.header" />
        </h1>
        <Tabs
          defaultActiveKey="1"
          type="card"
          tabBarStyle={{ borderBottom: 'none', overflow: 'hidden' }}
          >
          <TabPane
            tab={
              <span>
                <FormattedMessage id="NewsAndAlerts.news" />
              </span>
            }
            key="1">
            <NewsAndAlertsNews />
          </TabPane>

          <TabPane
            tab={
              <span>
                <FormattedMessage id="NewsAndAlerts.alerts" />
                <span className="count">12</span>
              </span>
            }
            key="2">
            <NewsAndAlertsAlerts />
          </TabPane>
        </Tabs>
      </div>
    );
  }
}
const mapDispatchToProps = dispatch => ({
  onUpdateLanguage (args) {
    dispatch(localeSet(args));
    return Promise.resolve();
  }
});

export default connect(mapDispatchToProps)(injectIntl(NewsAndAlerts));
