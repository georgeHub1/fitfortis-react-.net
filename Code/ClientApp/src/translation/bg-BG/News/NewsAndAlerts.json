{
  "header": "Новини & важни съобщения",
  "news": "Новини",
  "alerts": "Важни съобщения"
}
