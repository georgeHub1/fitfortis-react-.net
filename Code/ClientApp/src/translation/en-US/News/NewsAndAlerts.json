{
  "header": "News & Alerts",
  "news": "News",
  "alerts": "Alerts"
}
