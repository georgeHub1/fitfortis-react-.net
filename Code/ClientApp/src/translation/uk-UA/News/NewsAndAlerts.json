{
  "header": "Новини і сповіщення",
  "news": "Новини",
  "alerts": "Сповіщення"
}
