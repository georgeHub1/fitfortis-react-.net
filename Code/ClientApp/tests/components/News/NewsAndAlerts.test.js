import React from 'react';
import { shallow } from 'enzyme';

import NewsAndAlerts from '../../../src/components/News/NewsAndAlerts.js';

const Container = NewsAndAlerts;

describe('NewsAndAlerts', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Container/>);

    expect(wrapper).toMatchSnapshot();
  });
});
