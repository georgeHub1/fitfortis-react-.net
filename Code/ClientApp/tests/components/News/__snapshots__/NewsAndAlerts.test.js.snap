// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsAndAlerts renders correctly 1`] = `ShallowWrapper {}`;
